snippets/serial-console/snippet.yml

```diff
@@ -1,3 +1,16 @@
 name: serial-console
 append:
   EXTRA_CONF_FILE: serial-console.conf
+boards:
```

Comment:
don't think you need to explicitly mention the EXTRA_DTC_OVERLAY_FILE entries. Just put the overlays, with a file name matching the board name, in boards/ (see nordic-log-stm snippet for example)

w.r.t. enabling LOG_BACKEND_UART via the conf file, it doesn't feel like this belongs to this snippet (since it's a _serial console_ snippet)

Reply:
@kartben , thanks for pointing out that the overlay and conf files are automatically picked up when they have a matching name. Seems like the "Writing snippets" documentation would benefit from documenting this :)

About UART vs Serial, it sounds like the preferred approach would be to define a new snippet "uart-console", similar to "rtt-console", "ram-console", "nus-console" etc. Do you agree?

Reply:
> About UART vs Serial, it sounds like the preferred approach would be to define a new snippet "uart-console", similar to "rtt-console", "ram-console", "nus-console" etc. Do you agree?

not sure what you mean, it's already serial/uart console snippet? My point was that since it's a _console_ snippet, it probably shouldn't be messing with _log_ backend settings

Reply:
Do you have some suggestion on how to proceed with this one?

Reply:
I'm trying to recall why I left LOG_BACKEND_UART disabled by default in the nrf52_bsim, and I can't recall if there was any good reason (beyond "I'm used to see the LOG output in stdout for this target"). 
(But for the use case @rugeGerritsen wanted, the printk backend is the logger)
So maybe what we should just remove this line
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/native/nrf_bsim/nrf52_bsim_defconfig#L6

Reply:
I was gonna say, I'm sure there's a good reason it's disabled by default... but if there isn't then we might be all set :)
Along the same line, I would suggest having the chosen directly in the boards' DTS, this way you can basically just remove all the overlays :) (uart20 is already `okay`, fwiw).

Reply:
@aescolar , is there a reason for not setting the chosen property for these bsim boards?

Reply:
I don't think there is any good reason why they should not.
I think they were never added just because interactive use was never much in focus for these targets so we never got there.
So, sorry for not having realized and proposed that earlier :)

Reply:
No problem! I will create a separate PR to add this.